forem/forem
# conditionally render create post button for admins #16490

## What type of PR is this? (check all applicable)


## Description

Leverages the`limit_post_creation_to_admins` feature flag put in place by @jeremyf to conditionally render the "Create Post" button via a `<turbo-frame />` via Hotwire paradigm. When the feature is enabled, the "Create Post" button will dynamically render via role based access control (RBAC) as set forth in the `ArticlePolicy`. 

Closes #16490 

## QA Instructions, Screenshots, Recordings

**To Test:**
*  log in as an admin and create and/or enable `limit_post_creation_to_admins` feature flag.
* verify that the "Create Post" button is visible 
* log in as a non admin user with feature enabled
* verify that the "Create Post" button is not visible

More robust testing instructions can be found in the testing plan doc.

### UI accessibility concerns?

No as this only conditionally show/hides an existing element via RBAC.

## Added/updated tests?


## [Forem core team only] How will this change be communicated?



## [optional] What gif best describes this PR or how it makes you feel?

![](https://media.giphy.com/media/XOC9T1OK799kXV5iRS/giphy.gif)
